### DISCUSSION TITLE
Link doesn't reload page or reset state, it just swaps content??

### DISCUSSION THREAD

**Discussion by WillPwn4Food** [vercel/next.js#24183] [Help]
This seems simple, but perhaps the docs lack in explaining what is going on.

For context, this is a quiz site.
I've got a route **/quizzes/[quiz]/[question]** which for this example we'll say is **/quizzes/car-quiz/1**.

On this page, I see a question with 4 possible answers. I answer the question with a button click that flips the toggle on my useState for "showAnswer" from false to true, and the continue button appears along with if you were right or wrong.

The continue button is using 'next/link' and has the href="/quizzes/car-quiz/2&correct=1", generated on the button click.

I click the link, and the page **appears** to transition, except now on the new page '**/quizzes/car-quiz/2&correct=1**' it hasn't rerun anything on the page and even still has the same state in all my useState hooks, including "showAnswer" which is still true!

So here I am, on the next question page, and my scoreboard hasn't updated by reading from the queryParams for "correct" and I already see the answer and next continue button still shown, but with all the answer text for the next question (what a spoiler alert).

I've tried adding prefetch={false} to my link, but no difference. I've tried using router.push originally and had the same issue.
In the web console useEffect doesn't fire. I can see in my terminal that getStaticProps fired for the new page.

All my pages are statically generated using getStaticProps. I don't understand why it's not firing a refresh of things if I've navigated to a new page? Yes it's the same route, but the dynamic params have updated.

When it navigates to the new page, if it's even really doing that, I expect it to run all the code in the component again, including my variable declarations like: **const [showAnswer, setShowAnswer] = useState(false);**, but clearly it's not doing that. If I add a console.log it'll run that, but won't rerun my state functions?

I originally just used /quizzes/[...quiz] with shallow url updates to keep state and just fake update the url, but it wasn't capturing things in Google Analytics properly for page changes, so I thought I'd just break it up into separate actual pages, but now this is confusing the hell out of me!

Am I missing something here, what is going on?

**Comment by apollonian** [comment]
Has it got something to do with the URL of the next page? 
I believe it should be `/quizzes/car-quiz/2?correct=1` (query params begin with a "?").

**Reply by WillPwn4Food** [reply]
Sorry, I wrote it wrong in my description above, I did have **"?"** in my code. I even built the URLs using the object method:
```
var nextQuestionPath = {
            pathname: "/quizzes/[quiz]/[question]",
            query: {
                quiz: slug,
                question: nextQuestionNumber,
                correct: newCorrectCount,
            }
        };
```
